src/components/ChannelList/ChannelList.js

```diff
@@ -1,543 +1,361 @@
-      ) {
-        this.props.onChannelDeleted(this, e);
-      } else {
-        const channels = this.state.channels;
-        const channelIndex = channels.findIndex(
-          (channel) => channel.cid === e.channel.cid,
-        );
-
-        if (channelIndex < 0) return;
-
-        // Remove the deleted channel from the list.s
-        channels.splice(channelIndex, 1);
-        this.setState({
-          channels: [...channels],
-          channelUpdateCount: this.state.channelUpdateCount + 1,
-        });
-
-        // Unset active channel if it was the deleted one
-        if (e.channel.cid === this.props.channel.cid) {
-          this.props.setActiveChannel({});
-        }
-      }
+      return;
     }
 
-    if (e.type === 'channel.truncated') {
-      this.setState((prevState) => ({
-        channels: [...prevState.channels],
-        channelUpdateCount: prevState.channelUpdateCount + 1,
-      }));
-
-      if (
-        this.props.onChannelTruncated &&
-        typeof this.props.onChannelTruncated === 'function'
-      )
-        this.props.onChannelTruncated(this, e);
+    if (setActiveChannelOnMount) {
+      setActiveChannel(channels[0], watchers);
     }
-
-    return null;
   };
 
-  getChannel = async (type, id) => {
-    const channel = this.props.client.channel(type, id);
-    await channel.watch();
-    return channel;
+  // When channel list (channels array) is updated without any shallow changes (or with only deep changes), then we want
+  // to force the channel preview to re-render.
+  // This happens in case of event channel.updated, channel.truncated etc. Inner properties of channel is updated but
+  // react renderer will only make shallow comparison and choose tonot to re-render the UI.
+  // By updating the dummy prop - channelUpdateCount, we can force this re-render.
+  const forceUpdate = () => {
+    setChannelUpdateCount((count) => count + 1);
   };
 
-  moveChannelUp = (cid) => {
-    const channels = this.state.channels;
-    // get channel index
-    const channelIndex = this.state.channels.findIndex(
-      (channel) => channel.cid === cid,
-    );
-    if (channelIndex <= 0) return;
-
-    // get channel from channels
-    const channel = channels[channelIndex];
-
-    //remove channel from current position
-    channels.splice(channelIndex, 1);
-    //add channel at the start
-    channels.unshift(channel);
-
-    // set new channel state
-    this.setState({
-      channels: [...channels],
+  const [channelUpdateCount, setChannelUpdateCount] = useState(0);
+
+  const {
+    channels,
+    loadNextPage,
+    hasNextPage,
+    status,
+    setChannels,
+  } = usePaginatedChannels(
+    props.client,
+    props.filters,
+    props.sort,
+    props.options,
+    activeChannelHandler,
+  );
+
+  useMobileNavigation(channelListRef, props.navOpen, props.closeMobileNav);
+
+  // All the event listeners
+  !props.lockChannelOrder && useMessageNewListener(setChannels);
```

Comment:
This breaks the first rule of Hooks https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level.